### PR TITLE
fix: added npm install to gulp task

### PR DESCRIPTION
**Pull request by bc-zoharmuzafi**
#### What?
After starting to check in package-lock.json, we should adjust the gulp task for releasing to run npm install prior the push of the release to github.

#### Screenshots
screen shot 2019-02-27 at 2 20 06 pm

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -93,14 +93,20 @@ function deployWebpack() {
         .on('error', logError);
 }
 
+function installNpmDependencies() {
+    return gulp.src('package.json')
+        .pipe(exec('npm install'), logError)
```
Comment (lord2800) [inline gulpfile.js]: Does it make sense to check if the version of npm supports `ci` and if there's also a package-lock.json file, to use `npm ci` instead?

Comment (mattolson) [reply gulpfile.js]: If we use `npm install` at build time, we run the risk of making minor version changes at build time that were not validated by the author.

If we can't use `npm ci` to update the package version in `package-lock.json` (which would ensure that only the version number is updated and no dependencies), then I think we should modify the bump task to bump the version number in both `package.json` and `package-lock.json`.

File: gulpfile.js [hunk 2]
```diff
@@ -87,6 +91,21 @@ function bumpTask() {
         .on('error', logError);
 }
 
+function bumpPackageLock(targetVersion) {
+    if (!lockfileVersions.includes(packageLock["lockfileVersion"])) {
+        throw new Error("check lockfileVersions in package-lock.json");
```
Comment (mattolson) [inline gulpfile.js]: Should say `Release script only supports version ${supportedLockFileVersion}`

File: gulpfile.js [hunk 3]
```diff
@@ -10,6 +10,8 @@ const git = require('gulp-git-streamed');
 const gulp = require('gulp');
 const gulpif = require('gulp-if');
 const gutil = require('gulp-util');
+const lockfileVersions = [1];
```
Comment (mattolson) [inline gulpfile.js]: How about `supportedLockFileVersion`, and make it a single value rather than a list. If we get to the point that we need to support a new version, we'll likely need to re-evaluate this code anyway.

File: gulpfile.js [hunk 4]
```diff
@@ -10,6 +10,8 @@ const git = require('gulp-git-streamed');
...
+const packageLock = require('./package-lock.json');
```
Comment (mattolson) [inline gulpfile.js]: I would remove this and instead rely on `gulp.src`

File: gulpfile.js [hunk 5]
```diff
@@ -87,6 +91,21 @@ function bumpTask() {
...
+        throw new Error("check lockfileVersions in package-lock.json");
+    }
+    
+    return gulp.src('package-lock.json')
+        // bump the version number in those files
+        .pipe(bump({ version: targetVersion }))
+        // save it back to filesystem
+        .pipe(gulp.dest(process.cwd()))
+        // change last modified date
+        .pipe(exec('touch -c package.json'))
```
Comment (mattolson) [inline gulpfile.js]: Should be `package-lock.json`

File: gulpfile.js [hunk 6]
```diff
@@ -74,6 +76,8 @@ function bumpTask() {
             targetVersion = (res.type !== 'custom') ? semver.inc(currentVersion, res.type) : res['custom-version'];
             responses = res;
 
+            bumpPackageLock(targetVersion);
```
Comment (mattolson) [inline gulpfile.js]: There is a simpler approach to this. Look at the readme for gulp-bump: https://github.com/stevelacy/gulp-bump#bumping-version-and-outputting-different-files

File: gulpfile.js (hunk 4 above)
Comment (bc-zoharmuzafi) [reply gulpfile.js]: @mattolson we will need that outside the scope that use the src, to validate the lockfileVersion

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,13 +1,3 @@
-<a name="1.15.6"></a>
```
Comment (lord2800) [inline CHANGELOG.md]: Is this change intended?

File: package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bigcommerce/stencil-cli",
-  "version": "1.15.6",
```
Comment (lord2800) [inline package.json]: Is this change intended?

File: gulpfile.js (hunk 4 above)
Comment (lord2800) [reply gulpfile.js]: Do we need to read it in up here, or can we do that in `bumpTask`?

File: CHANGELOG.md (hunk 7 above)
Comment (bc-zoharmuzafi) [reply CHANGELOG.md]: @lord2800 yes, this is part or "rolling back" from the release that we made and re-release 1.15.6

File: package.json (hunk 8 above)
Comment (bc-zoharmuzafi) [reply package.json]: @lord2800 same as the CHANGELOG

File: gulpfile.js (hunk 4 above)
Comment (bc-zoharmuzafi) [reply gulpfile.js]: @lord2800 Can do it either way, we use it only in bumpTask anyway. I can change that if @mattolson will accept my previous comment related to not removing that completely.

File: CHANGELOG.md (hunk 7 above)
Comment (lord2800) [reply CHANGELOG.md]: Got it.

File: gulpfile.js [hunk 9]
```diff
@@ -10,6 +10,7 @@ const git = require('gulp-git-streamed');
 const gulp = require('gulp');
 const gulpif = require('gulp-if');
 const gutil = require('gulp-util');
+const supportedLockFileVersion = [1];
```
Comment (lord2800) [inline gulpfile.js]: 🍹 Since this is an array, it should probably be called `supportedLockFileVersions`.
